### ISSUE TITLE
HISAT2 in workflows: renaming output is broken

### ISSUE THREAD

**Issue by NCEichner** [galaxyproject/tools-iuc#1478]
Hallo, iuc-team!
HISAT2 has a problem with renaming outputs when the tool is used within workflows.

In two seperate workflows I use HISAT2 with either paired or single-end inputs (fastq-files e.g. "libname.R1" and "libname.R2"). The bam-output should be renamed to "libname.bam" and therfore I set the rule for rename output_alignments to "#{forward|basename}.bam" (similar to bowtie2). But I always get output named ".bam" (The name of the input-datafile is missing completely).
I played around with other variables from the workflow window (alone or in combination) and even tried to use variable names directly from the .xml-files (wrapper or macros). In some cases I also left out the "basename"-option but nothing changed - I always get ".bam" 
The automatic naming ("HISAT2 on data 2 and data1") works fine but then I have to rename all output files manually before starting the downstream analysis within a seperate workflow.
Would be nice if this could be fixed.

**Comment by mblue9** [comment]
Hi @NCEichner,

I took a look to see if the changes I've added in the PR to fix the strandedness bug will fix this issue, but unfortunately they won't :(

I think I can see what the problem is - it looks like nested inputs don't work with Rename Datasets?

In the Workflow Editor Rename Dataset section, for paired reads you get these "Valid Inputs" listed:

_for Bowtie2:_ 
library|input_1,  library|input_2
(renaming works using #{input_1|basename}.bam)

_for HISAT2:_ 
input_format|paired|collection|forward, input_format|paired|collection|reverse.
(renaming does not work using #{forward|basename}.bam)

_for HISAT2 new PR:_
reads|library|input_1, reads|library|input_2
(renaming does not work using #{input_1|basename}.bam, but does work if I remove the section called "reads", a section I added for cosmetic purposes)

So I think to fix this issue the options are:

- the HISAT2 wrapper needs to be changed to not use a nested input variable (that would make it work similarly to Bowtie2. But does Bowtie2 handle renaming of paired collections? it looks like it might not from here: https://biostar.usegalaxy.org/p/14911/)
- the Galaxy codebase needs to be changed (or maybe hack the code yourself like some people did here https://github.com/galaxyproject/galaxy/issues/1675)
- find another way to do the sample renaming 

If anyone in IUC has a suggestion would be great to hear as it would be really good to get this renaming outputs working.
